api/project_test.go

```diff
@@ -202,3 +202,55 @@ func TestFormatTemplate(t *testing.T) {
 		}
 	}
 }
+
+func TestGetBaseDatabaseName(t *testing.T) {
+	tests := []struct {
+		name         string
+		databaseName string
+		template     string
+		labelsJSON   string
+		want         string
+		errPart      string
+	}{
+		{
+			"tenant_label_success",
+			"db1_tenant123",
+			"{{DB_NAME}}_{{TENANT}}",
+			"[{\"key\":\"bb.location\",\"value\":\"us-central1\"},{\"key\":\"bb.tenant\",\"value\":\"tenant123\"},{\"key\":\"bb.environment\",\"value\":\"Dev\"}]",
+			"db1",
+			"",
+		},
+		{
+			"tenant_location_label_success",
+			"us-central1...db你好_tenant123",
+			"{{LOCATION}}...{{DB_NAME}}_{{TENANT}}",
+			"[{\"key\":\"bb.location\",\"value\":\"us-central1\"},{\"key\":\"bb.tenant\",\"value\":\"tenant123\"},{\"key\":\"bb.environment\",\"value\":\"Dev\"}]",
+			"db你好",
+			"",
+		},
+		{
+			"tenant_label_fail",
+			"db1_tenant123",
+			"{{DB_NAME}}_{{LOCATION}}",
+			"[{\"key\":\"bb.location\",\"value\":\"us-central1\"},{\"key\":\"bb.tenant\",\"value\":\"tenant123\"},{\"key\":\"bb.environment\",\"value\":\"Dev\"}]",
+			"",
+			"failed to find base database name",
```

Comment:
Can we make error more specific? IIUC, this one fails because of we don't find the location, the DB_NAME is actually there?

Reply:
This is the partial error errPart. Updated the error message to better indicate that this could only happen if the database name doesn't follow the template.